src/decorator/field-name.ts

```diff
@@ -0,0 +1,29 @@
+import { METADATA_DESERIALIZE_FIELD_NAME } from './deserialize-field-name';
+
+export const METADATA_CUSTOM_FIELDS = 'serializer:field:properties';
+/**
+ * Tags a property to map from a different field name.
+ *
+ * ## Example:
+ *
+ * ```typescript
+ * export class Example{
+ *      @FieldName('bar')
+ *      foo: string;
+ * }
+ *
+ * const result = serializer.deserialize<Example>({ bar: 'hey' }, Example);
+ * console.log(result.foo);
+ * ```
+ *
+ * This will print `hey` because the attribute with name `bar` has been mapped into the `foo` property.
+ *
+ * @decorator Property
+ */
+export function FieldName(fieldName: string): (...args: any[]) => void {
```

Comment:
The code looks to be the same as `DeserializeFieldName`. It would be worth it to factorise it into an external function. To make `FieldName` a combination of `DeserializeFieldName` and later of `SerializeFieldName`. It should be seen directly in the code that it's just a combination.